docs/registry/README.md

```diff
@@ -0,0 +1,385 @@
+
+Upon installation of an Event Source CR by a `Cluster Configurator`, the Source will register its EventTypes.
+
+Example:
+
+```yaml
+apiVersion: sources.eventing.knative.dev/v1alpha1
+kind: GitHubSource
+metadata:
+  name: github-source-sample
+  namespace: default
+spec:
+  eventTypes:
+    - push
+    - pull_request
+  ownerAndRepository: my-other-user/my-other-repo
+  accessToken:
+    secretKeyRef:
+      name: github-secret
+      key: accessToken
+  secretToken:
+    secretKeyRef:
+      name: github-secret
+      key: secretToken
+  sink:
+    apiVersion: eventing.knative.dev/v1alpha1
+    kind: Broker
+    name: default
+
+```
+ 
+By applying the above file, two EventTypes will be registered, with types `dev.knative.source.github.push` and 
+`dev.knative.source.github.pull_request`, source `my-other-user/my-other-repo`, for the `default` Broker in the `default`
+ namespace, and with owner `github-source-sample`.
+ 
+Note that the `Cluster Configurator` is the person in charge of taking care of authentication-related matters. E.g., if a new `Event Consumer` 
+wants to listen for events from a different GitHub repo, the `Cluster Configurator` will take care of the necessary secrets generation, 
+and new Source instantiation.
+ 
+In YAML, the above EventTypes would look something like these:
+
+```yaml
+apiVersion: eventing.knative.dev/v1alpha1
+kind: EventType
+metadata:
+  generateName: dev.knative.source.github.push-
+  namespace: default
+  owner: # Owned by github-source-sample
+spec:
+  type: dev.knative.source.github.push
+  source: my-other-user/my-other-repo
+  broker: default
+---
+apiVersion: eventing.knative.dev/v1alpha1
+kind: EventType
+metadata:
+  generateName: dev.knative.source.github.pullrequest-
+  namespace: default
+  owner: # Owned by github-source-sample
+spec:
+  type: dev.knative.source.github.pull_request
+  source: my-other-user/my-other-repo
+  broker: default
+```
+
+Two things to notice: 
+- We generate the names by stripping invalid characters from the original type (e.g., `_`)
+- The `spec.type` adds the prefix `dev.knative.source.github.` This is a **separate discussion** on whether we should 
+change the (GitHub) types or not.
+
+`1.2.` Manual Registration
+
+The `Cluster Configurator` manually `kubectl applies` an EventType CR.
+
+Example: 
+
+```yaml
+apiVersion: eventing.knative.dev/v1alpha1
+kind: EventType
+metadata:
+  name: repofork
+  namespace: default
+spec:
+  type: repo:fork
+  source: my-other-user/my-other-repo
+  broker: dev
+``` 
+
+This would register the EventType named `repofork` with type `repo:fork`, source `my-other-user/my-other-repo` 
+in the `dev` Broker of the `default` namespace.
+
+As under the hood, `kubeclt apply` just makes a REST call to the API server with the appropriate RBAC permissions, 
+the `Cluster Configurator` can give EventType `create` permissions to trusted parties, so that they can register 
+their EventTypes.  
+
+
+`1.3` Broker Auto Registration Policy
```

Comment:
Can we get rid of this option? It basically implies that at least one message of every event type will be dropped unless someone asks for all event types - which defeats the purpose of having a registry of event types people can choose from.

Also, we talked about how the Registry MUST be accurate and a full representation of what MAY flow thru the system. This option goes directly against that requirement by only seeing a subset

Reply:
We added this in case we don't know the EventTypes a Source (e.g., ContainerSource) may send. Let me leave it there for now, so that other people can also comment and maybe come up with a better solution, which might just be to remove it :) 
This needs further discussion.

Reply:
Having `source` as optional might be a good idea. I'm not sure if a list or a field would be better. With the list, I'm worried on how big an EventType can become, and it will probably be less efficient if we need to update it often. But I'm not sure.